lib/aws-manager.js

```diff
@@ -1245,50 +1120,26 @@ class AwsManager {
       assert(Array.isArray(states));
     }
 
-    let perRegionKills = {};
-
-    for (let region of this.ec2.regions) {
-      perRegionKills[region] = {
-        instances: [],
-        requests: [],
-      };
-    }
+    await Promise.all(_.map(this.ec2, async (ec2, region) => {
+      let instances = this.__apiState.instances.filter(instance => {
+        return instance.WorkerType === name && _.includes(states, instance.State.Name); 
+      });
 
-    for (let instance of this.__apiState.instances) {
-      if (instance.WorkerType === name && states.includes(instance.State.Name)) {
-        perRegionKills[instance.Region].instances.push(instance.InstanceId);
-      }
-    }
+      // _.concat just doesn't seem to work, so I'm doing this a slightly
+      // different way.  _.concat is showing as undefined :/
```

Comment:
I'm no `.concat` fan either.. regardless if it is `[].concat` or `_.concat`..

I suggest:
```js
let requests = _.flatten([
  this.__apiState.requests.filter(requests => {
    ....
  }),
  this.__internalState.filter(requests => {
    ....
  }),
]);
```

`_.flatten([1, [2], [3,4], [[5]]]) == [1, 2, 3, 4, [5]]` is pretty nice in that it creates a new list... rather than concatenating... And it only removes one level of arrays...